### PR TITLE
new toolbar version - v0.3.0

### LINKED ISSUE #78: Toolbar not clickable -- needs upgrade to "leaflet-draw-toolbar" package (next steps outlined)
In previous version of leaflet(0.7.7) toolbar is enabled when clicked on image but with the leaflet 1.2 version there are few issues. The click events are not registered in _initEvents(); , and when we click on image overlay this._showToolbar() is not invoking.

```
leaflet.js:5 Deprecated include of L.Mixin.Events: this property will be removed in future releases, please inherit from L.Evented instead. Error
    at y (file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet/dist/leaflet.js:5:2048)
    at Function.v.extend (file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet/dist/leaflet.js:5:14706)
    at file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet-toolbar/dist/leaflet.toolbar.js:1:60
    at file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet-toolbar/dist/leaflet.toolbar.js:1:5309 
```

### PR DESCRIPTION
**Pull request by jywarren**
fixes #78

### CODE REVIEW

**Comment by jywarren** [comment]
```
Loading "grunt-karma.js" tasks...ERROR
>> TypeError: Cannot read property 'prototype' of undefined
Warning: Task "karma:test" not found. Use --force to continue
```

**Comment by jywarren** [comment]
Hmm, i think `grunt-karma` is starting to become tough to use, at least in the version `^0.9.0` that we're using. Other people are seeing this problem: https://github.com/mulesoft/api-console/issues/275

But in manual testing, our problem is:

`leaflet.distortableimage.js:548 Uncaught TypeError: Cannot read property 'extend' of undefined`

That's this line:

```js
var EditOverlayAction = L.ToolbarAction.extend({
```

Looks like we need to follow instructions here: https://github.com/leaflet/Leaflet.Toolbar/wiki/Building-custom-toolbars

Here's an example of this being done: https://github.com/Leaflet/Leaflet.toolbar/blob/324c4d058e86f6d0451212ec103f80eb1c6bc286/test/src/ToolbarActionSpec.js#L13-L20

```js
		Action = LeafletToolbar.ToolbarAction.extend({
			options: {
				toolbarIcon: {
					html: 'Test Icon',
					className: 'my-toolbar-icon'
				}
			}
		});
```

**Comment by jywarren** [comment]
Popup also needed some refactoring. Pushed a new version.

**Comment by jywarren** [comment]
OK, well the latest version does not produce errors. But it also does not produce a popup. This apparently needs further debugging! Any help appreciated.

File: dist/leaflet.distortableimage.js [hunk 1]
```diff
@@ -637,29 +637,14 @@ var EditOverlayAction = L.ToolbarAction.extend({
...
-			removeHooks: function() {
-				var map = this._map;
-
-				map.removeLayer(this.toolbar);
-			}
-		});
-
-		return L.Toolbar.prototype._getActionConstructor.call(this, A);
-	}
+L.DistortableImage.EditToolbar = new LeafletToolbar.Popup({
```
Comment (justinmanley) [inline dist/leaflet.distortableimage.js]: This should be `LeafletToolbar.Popup.extend` rather than `new Leaflet.Toolbar.Popup`.

**Comment by justinmanley** [comment]
I encountered the same error as you with the karma tests. Updating all of the karma and test-related dependencies to the latest version fixed the error for me.

**Comment by jywarren** [comment]
Great, thanks, just updated things and got it to build and test.

**Comment by jywarren** [comment]
Couple things remaining --

* Tests show `^ 'LeafletToolbar' is not defined.` though not in local testing. 
* I think the toolbar is now being created, but it's not shown, and I don't know that any buttons/actions are being added to it. Maybe we need to add them differently than we had before; the code is here: https://github.com/publiclab/Leaflet.DistortableImage/blob/master/src/edit/DistortableImage.EditToolbar.js#L97-L103

Closer!

**Comment by jywarren** [comment]
Aha! Got the toolbar to show. But not in the right place yet. Offsets seem wrong. Almost there!

**Comment by jywarren** [comment]
screenshot 2017-10-27 at 10 43 11 am

**Comment by jywarren** [comment]
OK -- fixed, I believe. I've just centered the menu above the image, as I had some trouble getting it to track the mouse position. I think this works pretty nicely. 

screenshot 2018-01-04 at 4 07 45 pm

**Comment by jywarren** [comment]
Bumping npm version. Passing all tests. I'm going to merge this in and release, finally!

**Pull request merged**